### ISSUE TITLE
When disableEditor is true, and I use a Custom Authentication Token, Node-Red Admin API method "settings" fails to work

### ISSUE THREAD

**Issue by sebascarra** [node-red/node-red#2662]
- [ ] Node-RED version: 1.1.2
- [ ] Node.js version: 12.1.0
- [ ] npm version: 6.14.5
- [ ] Platform/OS: RHEL 7 Docker container running Node-Red
- [ ] Browser: Google Chrome

Hello, I have noticed the following behavior.
When I disable the editor by setting disableEditor: true in settings.js, and Custom Authentication Token functionality is implemented (new feature in Node-Red 1.1.0, using 1.1.2 now) the Admin API /settings method seems to break.

When disableEditor: **false**:

```
docker run -p 1880:1880 -e NODE_TLS_REJECT_UNAUTHORIZED=0 nodered:1.1.2
curl -H "Authorization: Bearer MY_TOKEN" http://192.168.56.100:1880/settings
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /settings</pre>
</body>
</html>
```
When disableEditor: **true**:

```
docker run -p 1880:1880 -e NODE_TLS_REJECT_UNAUTHORIZED=0 nodered-ui:1.1.2
curl -H "Authorization: Bearer MY_TOKEN" http://192.168.56.100:1880/settings
{"httpNodeRoot":"/","version":"1.1.2","user":{"username":"myusername","permissions":["*"]},"context":{"default":"memory","stores":["memory"]},"flowEncryptionType":"system","tlsConfigDisableLocalFiles":false,"editorTheme":{"projects":{"enabled":false},"languages":["de","en-US","ja","ko","zh-CN","zh-TW"]}}
```

Long story short, I cannot hit the /settings Admin API method.

I can hit methods such as /nodes just fine. I haven't tested all API methods, but settings is a pretty essential one and it's broken. I have not tested if it breaks if authentication is disabled, but I can if necessary.

Thank you for the help.

**Comment by knolleary** [comment]
Have you got your `disabledEditor` true/false the wrong way around? You say you hit an issue when setting it to `true`, but in the example output you've shared, the error appears under the `false` example.

If you set `disabledEditor` to `true` it disables all routes relating to the editor. `/settings` is part of the Editor API, not the Admin API - so it gets disabled as well. That is how it has always been - I don't believe the custom token handling introduced in 1.1.0 has changed that.

So what you describe is the expected behaviour. Why is `/settings` an essential endpoint if you have disabled the editor?

**Comment by sebascarra** [comment]
Hello @knolleary, first of all thanks for the input and for everything you do for Node-Red and its community.

I did write true/false the other way around. The first code snippet is case "true" and the second is case "false" (the default).

If /settings is part of the Editor API, everything makes sense, and if that is the desired behaviour, it is no longer an issue/bug. However, that is probably not well reflected in the docs. Here we can see that the /settings endpoint is described as part of the Admin API:

![image](https://user-images.githubusercontent.com/21203285/88704795-ff25a700-d0e4-11ea-92ae-452fb6669302.png)

If I go to the Editor API docs (for some reason the fonts are messed up in them), I do read that settings is an object that contains the runtime settings and I guess that's why the /settings endpoint just returns that object, and the object only exists if the editor API is enabled:
![image](https://user-images.githubusercontent.com/21203285/88705325-bc180380-d0e5-11ea-8b32-f6b083a0d3ea.png)

This one made me lose an hour or so and maybe other people too. If it is useful for the project, I can offer myself to make any changes required to the docs.

Thank you for all your hard work Nick.

**Comment by knolleary** [comment]
Having looked at this some more, I have come to the conclusion that `/settings` *should* be part of the Admin API as documented. So we'll need to move it over in the code.

But we need to make sure the info returned reflects the fact the editor is disabled and not expose any editor-specific settings.
